### PR TITLE
Set correct types and validation for var.logging_enable_components and var.monitoring_enable_components

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -179,13 +179,13 @@ variable "networking_mode" {
 }
 
 variable "logging_enable_components" {
-  type        = string
-  description = "(Optional) The GKE components exposing logs. Supported values include: 'SYSTEM_COMPONENTS' and 'WORKLOADS'."
+  type        = set(string)
```
Comment (kassianh) [inline variables.tf:182]: The schema says this is a list of max items 1. `main.tf` treats it as a list of max items one, but should it? It seems confusing to do it this way as opposed to making the variable a string. What do you think @soerenmartius

**Review by kassianh — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (soerenmartius) [reply variables.tf:182]: Hey, `enable_components` in the `logging_config` block accepts a list, the `logging_config` block itself can only exist once

Comment (kassianh) [reply variables.tf:182]: Oh i see, you're right!

**Review by kassianh — APPROVED** [review]
lgtm

**Comment by soerenmartius** [comment]
resolved conflicts, rebased from `main`

**Review by czerasz-mineiros — APPROVED** [review]

**Pull request merged**